### PR TITLE
[debugcounterorch] check if counter type is supported before querying…

### PR DESCRIPTION
**Pull request by stepanblyschak**
… object availability count

Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>

DEPENDS: https://github.com/Azure/sonic-sairedis/pull/842
ALSO: https://github.com/Azure/sonic-sairedis/pull/843

**What I did**

Implemented check for debug counter type support prio to querying unsupported counter type availablility counter. 

**Why I did it**
This is done in order to prevent error message seen on mellanox platform:
```
Mar 11 16:28:42.945332 ptr-sonic-n2-t3 ERR syncd#SDK: [SAI_DEBUG_COUNTER.ERR] mlnx_sai_debug_counter.c[324]- mlnx_debug_counter_availability_get: Unsupported debug counter type - 1
Mar 11 16:28:42.948092 ptr-sonic-n2-t3 ERR syncd#SDK: [SAI_DEBUG_COUNTER.ERR] mlnx_sai_debug_counter.c[324]- mlnx_debug_counter_availability_get: Unsupported debug counter type - 0
```

**How I verified it**
1. Run on mellanox and observe no errors.
2. Run VS test for debug counters:
```                                                                                                                 
tests/test_drop_counters.py::TestDropCounters::test_deviceCapabilitiesTablePopulated PASSED               [  6%]
tests/test_drop_counters.py::TestDropCounters::test_flexCounterGroupInitialized PASSED                    [ 12%]
tests/test_drop_counters.py::TestDropCounters::test_createAndRemoveDropCounterBasic PASSED                [ 18%]
tests/test_drop_counters.py::TestDropCounters::test_createAndRemoveDropCounterReversed PASSED             [ 25%]
tests/test_drop_counters.py::TestDropCounters::test_createCounterWithInvalidCounterType PASSED            [ 31%] 
tests/test_drop_counters.py::TestDropCounters::test_createCounterWithInvalidDropReason PASSED             [ 37%]
tests/test_drop_counters.py::TestDropCounters::test_addReasonToInitializedCounter PASSED                  [ 43%]
tests/test_drop_counters.py::TestDropCounters::test_removeReasonFromInitializedCounter PASSED             [ 50%]
tests/test_drop_counters.py::TestDropCounters::test_removeAllDropReasons PASSED                           [ 56%]
tests/test_drop_counters.py::TestDropCounters::test_addDropReasonMultipleTimes PASSED                     [ 62%]
tests/test_drop_counters.py::TestDropCounters::test_addInvalidDropReason PASSED                           [ 68%]
tests/test_drop_counters.py::TestDropCounters::test_removeDropReasonMultipleTimes PASSED                  [ 75%]
tests/test_drop_counters.py::TestDropCounters::test_removeNonexistentDropReason PASSED                    [ 81%]
tests/test_drop_counters.py::TestDropCounters::test_removeInvalidDropReason PASSED                        [ 87%]
tests/test_drop_counters.py::TestDropCounters::test_createAndDeleteMultipleCounters PASSED                [ 93%]
tests/test_drop_counters.py::test_nonflaky_dummy PASSED                                                   [100%]
```

**Details if related**

### CODE REVIEW

File: orchagent/debug_counter/drop_counter.cpp [hunk 1]
```diff
@@ -342,6 +342,61 @@ unordered_set<string> DropCounter::getSupportedDropReasons(sai_debug_counter_att
...
+    unordered_set<string> supportedCounterTypes;
+
+    sai_s32_list_t enumValuesCapabilities;
+    vector<int32_t> saiCounterTypes;
+
+    const auto* meta = sai_metadata_get_attr_metadata(SAI_OBJECT_TYPE_DEBUG_COUNTER,
+                                                      SAI_DEBUG_COUNTER_ATTR_TYPE);
+    if (!meta)
+    {
+        SWSS_LOG_THROW("SAI BUG: metadata null pointer returned by "
```
Comment (prsunny) [inline orchagent/debug_counter/drop_counter.cpp:360]: Please specify ERROR log and return. No need to throw here. Same for below case as well

Comment (stepanblyschak) [reply orchagent/debug_counter/drop_counter.cpp]: Done

**Comment by liat-grozovik** [comment]
@prsunny  could you please review following the fix provided?

**Comment by stepanblyschak** [comment]
Does pipelines run with latest VS lib implementation?

**Comment by liat-grozovik** [comment]
@prsunny can you please review and approve?

**Review by prsunny — APPROVED** [review]

**Comment by prsunny** [comment]
@qiluo-msft, @liat-grozovik , I'm not sure why this alone was marked for 202012. Is this required to be backported? If so, I see that there are dependent PRs in sairedis.

**Comment by qiluo-msft** [comment]
@liat-grozovik Please let me know if this is not required for 202012.

**Pull request merged**